wherehows-dao/src/main/java/wherehows/dao/table/DictDatasetDao.java

```diff
@@ -30,4 +47,126 @@ public DictDataset findByUrn(String urn) {
   public DictDataset findById(int datasetId) {
     return findBy(DictDataset.class, "id", datasetId);
   }
+
+  /**
+   * Insert or update dict dataset table given information from MetadataChangeEvent
+   * @param identifier DatasetIdentifier
+   * @param auditStamp ChangeAuditStamp
+   * @param property DatasetProperty
+   * @param schema DatasetSchema
+   * @param deployments List<DeploymentDetail>
+   * @param tags List<String>
+   * @param capacities List<Capacity>
+   * @return dataset id
+   * @throws Exception
+   */
+  public DictDataset insertUpdateDataset(DatasetIdentifier identifier, ChangeAuditStamp auditStamp,
+      DatasetProperty property, DatasetSchema schema, List<DeploymentDetail> deployments, List<String> tags,
+      List<Capacity> capacities, PartitionSpecification partitions) throws Exception {
+
+    String urn = toWhDatasetUrn(identifier);
+
+    // find dataset if exist
+    DictDataset dataset = null;
+    try {
+      dataset = findBy(DictDataset.class, "urn", urn);
+    } catch (Exception e) {
+      log.debug("Can't find dataset " + urn, e.toString());
```

Comment:
log.warn

Reply:
I don't think you need e.toString here? It automatically print out the exception full name.

Reply:
This is for testing if the dataset exist, and for all new datasets, this will throw exception. Log.warn might give us too many false alarms. Changing to log.info